lib/ansible/modules/cloud/digital_ocean/digital_ocean_sshkey_facts.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'metadata_version': '1.1'}
+
+
+DOCUMENTATION = '''
+---
+module: digital_ocean_sshkey_facts
+short_description: DigitalOcean SSH keys facts
+description:
+     - Fetch DigitalOcean SSH keys facts.
+version_added: "2.5"
+author: "Patrick Marques (@pmarques)"
+options:
+  oauth_token:
+    description:
+     - DigitalOcean API token.
+    required: true
+
+notes:
+  - Version 2 of DigitalOcean API is used.
+requirements:
+  - "python >= 2.6"
+'''
+
+
+EXAMPLES = '''
+- name: "List all sshkeys"
+  digital_ocean_sshkey_facts:
+    register: result
```

Comment:
Also could you add to the example... I think something like this would be valuable for users to see. Hopefully it will help explain when to use this module over the `digital_ocean_sshkey` module.

```
    - digital_ocean_sshkey_facts:
         oauth_token: "{{ my_do_key }}"

    - set_fact:
        pubkey: "{{ item.public_key }}"
      with_items: "{{ ssh_keys|json_query(ssh_pubkey) }}"
      vars:
        ssh_pubkey: "[?name=='ansible_crtl']"

    - debug:
        msg: "{{ pubkey }}"
```

I'm not sure the `register` command provides much value since your returning ansible_facts. Users can directly access `ssh_keys` as an ansible fact but this module spits out an object that will need a filter applied to it. Still useful if your trying to get a pub-key for additional operations on your servers. Might be helpful when seeding more keys on already running droplets.

Reply:
@BondAnthony Nice example, I changed it as you suggested